### DISCUSSION TITLE
Install all the optional groups

### DISCUSSION THREAD

**Discussion by le-chartreux** [python-poetry/poetry#7605] [Q&A] [answered]
Firstly, thank you for making poetry, that's a very cool project!

## Question

Is there a way to install all the optional groups specified in the pyproject.toml?
Something similar to `poetry install --all-extras` but for groups, e.g. `poetry install --with-all`.

## Motivation

I use optional groups to minimize the size of the virtual envs.

I use `nox` to automate my testing in multiple Python versions. Each session does one task (e.g. running `flake8`), so there are a lot of sessions. Since each nox's session runs in its own virtual env, there are a lot of virtual envs created. To minimize their size, I set to optional each dev dependency that isn't used by a file (see the detailed situation part): for small libraries like `flake8` it may seem useless, but for larger ones like `mypy` it's cool.

My problem is: to be able to execute directly from `poetry` an optional dependencies  (e.g. running `mypy --help`) I have to enter manually each of the optional groups when I install my poetry project: `poetry install --with flake8 --with black --with [...]`. It will be much practice to be able to run something like `poetry install --with-all`.

## Detailed situation

I have this organization for dependencies: 

- **Dependencies required for the app** are in `tool.poetry.dependencies`. E.g. `requests`.
- **Dev dependencies imported by some Python files** (that aren't part of the app) are grouped by themes in non-optional groups. These groups are not optional to be able to analyse the files that use them. E.g. the group `pytest` contains `pytest` and `pytest-cov`. The test files import members of the `pytest` group, but test files aren't part of the app.
- **Dev dependencies not imported by some Python files** are grouped by themes in optional groups. E.g. the group `flake8` contains `flake8` plus all its plugins, it does not contain anything required to run any file of the project, so it is optional.

**Comment by dimbleby** [comment, marked as answer]
#5461

**Reply by le-chartreux** [reply]
Thank you for letting me know that this is a discussed subject, but is there any plan to merge it to the master branch? It's still waiting since April.

**Reply by dimbleby** [reply]
I'm afraid that languishing unloved is all too common a fate for MRs in this project.  

I don't know whether anyone has plans to merge it, but all those months of inactivity suggest not.

**Reply by le-chartreux** [reply]
OK, pretty sad. Thank you :+1:

**Comment by schneebuzz** [comment]
#9744
